### ISSUE TITLE
OS X 10.10.3 - Computer sleep settings still put the system to sleep

### ISSUE THREAD

**Issue by mag01** [integralpro/nosleep#8]
OS X 10.10.3
11" MBA 2012
NoSleep 1.3.3 and 1.4.0 (tried both): Never sleep on Battery: enabled
System Preferences: Energy Saver: Battery:
- Computer sleep: 10 minutes
- Display Sleep: 2 minutes
- Put hard disks to sleep when possible: enabled

When running on battery, the computer still goes to sleep after the specified 10 minutes.
NoSleep doesn't prevent that from happening. It only prevents it from going to sleep when the lid is being closed.
Is this correct behaviour? I believe it should prevent entering the sleep mode completely.

Now I'm not really sure, but I have a feeling it behave correctly (e.g. prevented the computed from going to sleep completely) in OS X 10.8 and perhaps 10.9 as well.

**Comment by Quantumpanda** [comment]
OS X 10.10.3, 17" MBP Mid-2010, NoSleep 1.4.0

I am seeing the same behavior since upgrading to 10.10.3 from 10.9.5, but I'm not certain it's NoSleep that is to blame. I have always had to use Caffeine with NoSleep to completely prevent sleep—Caffeine prevents automatic timed sleep, and NoSleep keeps the lid closing from making it sleep. I came here assuming that the failure to prevent sleep was a problem with NoSleep, but after reading your notes, I realized that I've always had to combine it with Caffeine to get the full desired behavior. Caffeine is working properly when the lid is open—I can leave the machine open, untouched, and plugged in all day without it ever going into sleep or screensaver. But after I close the lid, timed automatic sleep does kick in, despite both Caffeine and NoSleep being active and enabled.

I'm hoping the developer is reading these and can shed some light on the matter. I don't really want to have to switch to an alternative program—Caffeine and NoSleep in combination have always just plain worked, each doing their respective parts of the job perfectly, whereas the alternates I've tried fail at one task or another, and they don't get along with having another app that does the same thing running to fill in the gap(s).

**Comment by vprasanth** [comment]
+1

**Comment by jpage4500** [comment]
I noticed this too.. I was using Caffeine before but switched to NoSleep because I wanted a way to keep my macbook air running with the display off. 

Note - I didn't shut the laptop lid but have a hotspot corner that puts turns the display off.

It's good to know using both apps works - but of course I'd rather just use 1 app that offers the most options.

**Comment by mele113** [comment]
After much personal research on These similar types of apps~~ My choice ~~ is http://semaja2.net/2014/10/insomniax-version-2-1-8-the-quiet-one/ ore

**Comment by mag01** [comment]
Yes I've been using NoSleep+Amphetamine for a while now, but InsomniaX looks interesting as it might combine finctionality of them both. I'll give it a try, thanks for a tip.

**Comment by staticfish** [comment]
Did anyone find a solution to this? It's happening for me on Caffeine, Owly, InsomniaX, KeepingYouAwake (my personal favourite) on 10.11.4 (2015 13' MBP).

It then must be a system setting that alters the caffeinate command - that or apple disabled it for either the hardware or OSX.

**Comment by amitasthana** [comment]
Its December 2017, I am latest MacOS New Sierra. Hardware is old from mid 2012. So definitely its not hardware issue.
 I am trying to use all the apps, command lines and one script, nothing working to keep my mac awake while lid is closed.
One thing I have noticed, when I keep the power plugged in, it keeps the mac awake on lid close and as soon as I remove the power cable, the mac goes to sleep.
I know this as I am using the HDMI and a external screen.
